File: 1646-get-maximum-in-generated-array/1646-get-maximum-in-generated-array.js
/**
 * @param {number} n
 * @return {number}
 */
var getMaximumGenerated = function(n) {
    if(n==0) return 0;
    const nums = [0,1];
    for(let i=2;i<=n;i++){
        if(i%2===0){
            nums[i] = nums[i/2];
        } else {
            nums[i] = nums[Math.floor(i/2)] + nums[Math.ceil(i/2)];
        }
        
    }

    return Math.max(...nums);
};

File: 1646-get-maximum-in-generated-array/1646-get-maximum-in-generated-array.py
class Solution:
    def getMaximumGenerated(self, n: int) -> int:
        if n==0: return 0
        
        nums = [0,1]
        biggest = 1
        for i in range(2,n+1):
            if i%2==0:
                nums.append(nums[i//2])
            else:
                new_num = nums[i//2] + nums[(i//2) + 1]
                nums.append(new_num)
                biggest = max(biggest,new_num)
        
        return biggest